compiler/plc_project/src/build_config.rs

```diff
@@ -64,6 +79,27 @@ impl ProjectConfig {
 
         Ok(project)
     }
+
+    fn validate(&self) -> Result<(), Diagnostic> {
+        let schema =
+            serde_json::from_str(crate::build_description_schema::PLC_JSON_SCHEMA).expect("A valid schema");
+        let instance = json!(self);
```

Comment:
did we just deserialize and reserialize the same file here?

Reply:
We deserialize and compile the schema here. There is re-serialization happening, but on the `ProjectConfig` struct since we need the json data to be of type `Object`. I initially tried to use the text-file as is, but that locks out additional validations like `additionalProperties` and `minLength`.